### ISSUE TITLE
Document requirement for babel-plugin-transform-es2015-modules-commonjs

### ISSUE THREAD

**Issue by gajus** [react-cosmos/react-cosmos#234]
It is not clear that `babel-plugin-transform-es2015-modules-commonjs` is required.

Note that webpack@2 does not need the latter module. Therefore, most webpack@2 users will not have it.

**Comment by skidding** [comment]
Are you referring to the fact that webpack@2 users will have es2015 modules disabled in their `.babelrc` (and let webpack handle es2015 modules), like this?
```js
{
  "presets": [
    ["es2015", { "modules": false }]
  ]
}
```
The react-cosmos modules are already published as es5, so the only problem I can anticipate is disabling Babel modules and using es2015 `import` inside `cosmos.config.js`, which isn't processed through webpack. Is this the problem you identified or something different?

**Comment by gajus** [comment]
> Are you referring to the fact that webpack@2 users will have es2015 modules disabled in their .babelrc (and let webpack handle es2015 modules), like this?

Yes.

> The react-cosmos modules are already published as es5, so the only problem I can anticipate is disabling Babel modules and using es2015 import inside cosmos.config.js, which isn't processed through webpack. Is this the problem you identified or something different?

I have a component, e.g. `src/components/TestComponent.js`

```js
import React from 'react';

export () => {
  return <div>Hello, World!</div>;
};
```

Suppose my `.babelrc` is:

```json
{
  "presets": [
    ["es2015", { "modules": false }]
  ]
}
```

With this setup, if I use `react-cosmos`, I am getting an error on `import React from 'react';` in the `src/components/TestComponent.js`.

I am assuming this is because `react-cosmos` directly depends on `webpack@1`?

https://github.com/react-cosmos/react-cosmos/blob/81ad59e73911faf8117ba3f02fb6786b316e2963/package.json#L74

**Comment by skidding** [comment]
> I am assuming this is because react-cosmos directly depends on webpack@1?

Yes, `react-cosmos-webpack` more specifically, has been built for and tested with `webpack@1`. We could make this explicit in the [Requirements](https://github.com/react-cosmos/react-cosmos/blob/81ad59e73911faf8117ba3f02fb6786b316e2963/README.md#requirements), but at the same time it would be great to find a way to make Cosmos work with both webpack 1 and 2. Publishing separate versions of `react-cosmos-webpack` for each webpack version is undesirable. 

One solution would be to try to upgrade Cosmos to use `webpack@2` internally and tell users when to turn off `babel-plugin-transform-es2015-modules-commonjs`. Another would be to move webpack to `peerDependencies` and rely on the user's webpack copy (provided that APIs match, of course). What do you think?

**Comment by gajus** [comment]
> One solution would be to try to upgrade Cosmos to use webpack@2 internally and tell users when to turn off babel-plugin-transform-es2015-modules-commonjs.

Sounds reasonable.

> Another would be to move webpack to peerDependencies and rely on the user's webpack copy (provided that APIs match, of course). What do you think?

This is preferred.

**Comment by skidding** [comment]
Fixed via #262
